### PR TITLE
Vapordex

### PR DESCRIPTION
**Pull request by royvardhan**

### CODE REVIEW

File: dexs/vapordex/index.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  const cumulativeUniqueUsers = response.protocols[0].cumulativeUniqueUsers;
+
+  // Calculating the total length of daily snapshots to determine the latest one
+  const responseLength = response.financialsDailySnapshots.length;
+
+  // Daily volume in USD
+  const latestDailyVolumeUSD = Math.round(
+    response.financialsDailySnapshots[responseLength - 1].dailyVolumeUSD
+  );
+  return {
```
Comment (dtmkeng) [inline dexs/vapordex/index.ts:52]: return function fetch dexs have interface
```ts
timestamp: number; 
dailyVolume?: string;
totalVolume?: string;

```

File: dexs/vapordex/index.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    numRes.push(parseInt(res[i]));
+  }
+  const TVL = numRes.reduce((partialSum, a) => partialSum + a, 0);
+  return TVL;
+};
+
+const START_TIME = 1663584236;
+
+const fetch = async () => {
+  const response = await getGQLClient().request(getData());
```
Comment (dtmkeng) [inline dexs/vapordex/index.ts:33]: Do you have query by timestamp for backfill your volume data ?

File: dexs/vapordex/index.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    cumulativeVolume: `${cumulativeVolumeUSD}`,
+    cumulativeUniqueUsers: `${cumulativeUniqueUsers}`,
+  };
+};
+
+const adapter = {
+  adapter: {
+    [CHAIN.AVAX]: {
+      fetch: async () => fetch(),
+      start: async () => START_TIME,
```
Comment (dtmkeng) [inline dexs/vapordex/index.ts]: if your adapter. does not have function backfill you should set flag `runAtCurrTime: true`

**Comment by dtmkeng** [comment]
could you delete `.DS_store`

**Comment by royvardhan** [comment]
> could you delete `.DS_store`

Yup, deleted!

**Comment by royvardhan** [comment]
@dtmkeng Hi, is there anything needed from outside?

File: dexs/vapordex/index.ts (hunk 1 above)
Comment (royvardhan) [reply dexs/vapordex/index.ts:52]: Does that mean it should not return `tvl` and `uniqueUsers`

Comment (dtmkeng) [reply dexs/vapordex/index.ts:52]: yes

Comment (dtmkeng) [reply dexs/vapordex/index.ts:52]: for tvl you should open pr on https://github.com/DefiLlama/DefiLlama-Adapters

**Comment by royvardhan** [comment]
@dtmkeng changes pushed as requested. Pls review

**Comment by llamatester** [comment]
The vapordex adapter exports: 
        
 
 _______________________________________
    Dexs for 7/12/2022
    _______________________________________
    
    AVAX 👇
    Backfill start time: 1/1/1970
    Time stamp: 1670509244
    Daily volume: 43
    Total volume: 189138
    
    
    
    
    Done in 4.56s.

**Pull request merged**